src/tokenizer.cpp

```diff
@@ -293,10 +293,10 @@ static void cancel_token(Tokenize *t) {
 }
 
 static void end_float_token(Tokenize *t) {
-    if (t->radix == 10) {
+    if (t->radix == 10 || t->radix == 16) {
         uint8_t *ptr_buf = (uint8_t*)buf_ptr(t->buf) + t->cur_tok->start_pos;
         size_t buf_len = t->cur_tok->end_pos - t->cur_tok->start_pos;
-        if (bigfloat_init_buf_base10(&t->cur_tok->data.float_lit.bigfloat, ptr_buf, buf_len)) {
+        if (bigfloat_init_buf(&t->cur_tok->data.float_lit.bigfloat, ptr_buf, buf_len)) {
             t->cur_tok->data.float_lit.overflow = true;
         }
         return;
```

Comment:
We can remove the proceeding following float parsing code now, right? Only radix 10 and 16 is valid for float literals.

Reply:
Ahh my mistake, just noticed we accept octal and binary float literals. Interesting, didn't know that.

Reply:
I did that at first but it broke tests and so I put it back and filed #2093